### PR TITLE
[MODTLR-5] Add /tlr/ecs-tlr POST endpoint

### PR DESCRIPTION
**Pull request by roman-barannyk**
## Purpose
Add /tlr/ecs-tlr POST endpoint

Resolves: [MODTLR-5](https://issues.folio.org/browse/MODTLR-5)

### CODE REVIEW

File: src/main/java/org/folio/controller/EcsTlrController.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+      .orElseGet(() -> ResponseEntity.notFound().build());
+  }
+
+  @Override
+  public ResponseEntity<EcsTlr> postEcsTlr(EcsTlr ecsTlr) {
+    log.debug("postEcsTlr:: parameters ecsTlr: {}", ecsTlr);
+
+    try {
+      return new ResponseEntity<>(ecsTlrService.post(ecsTlr), CREATED);
+    } catch (Exception e) {
```
Comment (OleksandrVidinieiev) [inline src/main/java/org/folio/controller/EcsTlrController.java:40]: Do we really need to catch errors here? I believe all exceptions are converted into 500 by default anyway...
Also, `ResponseEntity` has a neat static constructor: `ResponseEntity.status(...).body(...)`

File: src/main/java/org/folio/domain/entity/EcsTlrEntity.java [hunk 2]
```diff
@@ -10,21 +14,29 @@
...
-@Table(name = "title_level_request")
+@Table(name = "ecs_tlr")
 @Getter
 @Setter
 @ToString
 @AllArgsConstructor
 @NoArgsConstructor
 @Builder
-public class TitleLevelRequestEntity {
+public class EcsTlrEntity {
```
Comment (OleksandrVidinieiev) [inline src/main/java/org/folio/domain/entity/EcsTlrEntity.java:26]: `@Getter @Setter @ToString` can be replaced with `@Data`

File: src/main/java/org/folio/service/impl/EcsTlrServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+@RequiredArgsConstructor
+@Log4j2
+public class EcsTlrServiceImpl implements EcsTlrService {
+
+  private final EcsTlrRepository ecsTlrRepository;
+  private final EcsTlrMapper requestsMapper;
+
+  @Override
+  public Optional<EcsTlr> get(UUID id) {
+    return ecsTlrRepository.findById(id)
```
Comment (OleksandrVidinieiev) [inline src/main/java/org/folio/service/impl/EcsTlrServiceImpl.java:25]: add logging?

File: src/main/resources/swagger.api/ecs-tlr.yaml [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/ecs-tlr"
+        required: true
+      responses:
+        '201':
+          $ref: "#/components/responses/ecs-tlr"
+        '400':
```
Comment (OleksandrVidinieiev) [inline src/main/resources/swagger.api/ecs-tlr.yaml:19]: Our controller does not return 400 (yet?)

File: src/main/resources/swagger.api/ecs-tlr.yaml [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+  /tlr/ecs-tlr/{requestId}:
+    get:
+      description: Retrieve ECS TLR by id
+      operationId: getEcsTlrById
+      parameters:
+        - $ref: '#/components/parameters/requestId'
+      responses:
+        '200':
+          $ref: '#/components/responses/ecs-tlr'
+        '400':
```
Comment (OleksandrVidinieiev) [inline src/main/resources/swagger.api/ecs-tlr.yaml:32]: same here

File: src/main/resources/swagger.api/ecs-tlr.yaml [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+      schema:
+        type: string
+        format: uuid
+  responses:
+    ecs-tlr:
+        description: ECS TLR object
+        content:
+          application/json:
+            schema:
+              $ref: 'schemas/EcsTlr.yaml#/EcsTlr'
```
Comment (OleksandrVidinieiev) [inline src/main/resources/swagger.api/ecs-tlr.yaml:63]: wrong indentation

File: src/main/resources/swagger.api/schemas/EcsTlr.yaml [hunk 7]
```diff
@@ -0,0 +1,35 @@
+EcsTlr:
+  description: ECS TLR
+  type: "object"
+  properties:
+    id:
+      description: "Id of the ecs-tlr"
+      $ref: "uuid.yaml"
+    instanceId:
+      description: "ID of the instance being requested"
```
Comment (OleksandrVidinieiev) [inline src/main/resources/swagger.api/schemas/EcsTlr.yaml:9]: Let's make it uniform: Id or ID

File: src/test/java/org/folio/service/EcsTlrServiceTest.java [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+
+@ExtendWith(MockitoExtension.class)
+class EcsTlrServiceTest {
+
+  @InjectMocks
+  private EcsTlrServiceImpl ecsTlrService;
+  @Mock
+  private EcsTlrRepository ecsTlrRepository;
+  @Mock
+  private EcsTlrMapper ecsTlrMapper;
```
Comment (OleksandrVidinieiev) [inline src/test/java/org/folio/service/EcsTlrServiceTest.java:30]: Do we have to mock the mapper here? Can we use a real one?

File: src/main/java/org/folio/domain/entity/EcsTlrEntity.java [hunk 9]
```diff
@@ -10,21 +14,29 @@
...
   @Convert(converter = UUIDConverter.class)
   private UUID id;
-
+  @Convert(converter = UUIDConverter.class)
+  private UUID instanceId;
+  @Convert(converter = UUIDConverter.class)
+  private UUID requesterId;
+  private String requestType;
+  private String requestLevel;
+  private String requestExpirationDate;
```
[the remark this answers is not in the document]
Comment (roman-barannyk) [reply src/main/java/org/folio/domain/entity/EcsTlrEntity.java]: Changed to Date, as it is used in mod-circulation-storage

File: src/main/java/org/folio/domain/entity/EcsTlrEntity.java (hunk 2 above)
Comment (roman-barannyk) [reply src/main/java/org/folio/domain/entity/EcsTlrEntity.java:20]: updated

File: src/main/java/org/folio/service/impl/EcsTlrServiceImpl.java (hunk 3 above)
Comment (roman-barannyk) [reply src/main/java/org/folio/service/impl/EcsTlrServiceImpl.java:27]: added

File: src/main/resources/swagger.api/ecs-tlr.yaml (hunk 4 above)
Comment (roman-barannyk) [reply src/main/resources/swagger.api/ecs-tlr.yaml:19]: Our controller currently does not directly return a 400 response, however, the Swagger schema handles the validation and returns a 400 response if the incoming request is deemed invalid

File: src/main/resources/swagger.api/schemas/EcsTlr.yaml (hunk 7 above)
Comment (roman-barannyk) [reply src/main/resources/swagger.api/schemas/EcsTlr.yaml]: fixed

File: src/test/java/org/folio/service/EcsTlrServiceTest.java (hunk 8 above)
Comment (roman-barannyk) [reply src/test/java/org/folio/service/EcsTlrServiceTest.java]: updated, thanks

**Review by alexanderkurash — APPROVED** [review]

**Review by OleksandrVidinieiev — APPROVED** [review]

**Pull request merged**